integrity/integration_test.go

```diff
@@ -207,5 +295,533 @@ func TestTypedIntegration_GetPut(t *testing.T) {
 		val, ok := result.Value.Get()
 		require.True(t, ok)
 		assert.Equal(t, value, val)
+
+		// Cleanup.
+		cleanupTyped(t, typed, "my-object")
+	})
+}
+
+func TestTypedIntegration_Delete(t *testing.T) {
+	executeOnStorage(t, func(t *testing.T, driverInstance driver.Driver) {
+		t.Helper()
+
+		fillDataOptions := FillDataOptions{
+			Prefix: "/test",
+			Names:  []string{"my-object"},
+			Records: []IntegrationStruct{
+				{
+					Name:  "test",
+					Value: 42,
+				},
+			},
+			Hashers: nil,
+			Signers: nil,
+		}
+
+		fillData(t, driverInstance, fillDataOptions)
+
+		ctx := t.Context()
+		storageInstance := storage.NewStorage(driverInstance)
+		typed := integrity.NewTypedBuilder[IntegrationStruct](storageInstance).
+			WithPrefix("/test").
+			Build()
+
+		err := typed.Delete(ctx, "my-object")
+		require.NoError(t, err, "Delete should succeed")
+
+		_, err = typed.Get(ctx, "my-object")
+		require.Error(t, err, "Get should fail")
+	})
+}
+
+func TestTypedIntegration_Get_EmptyStorage(t *testing.T) {
+	executeOnStorage(t, func(t *testing.T, driverInstance driver.Driver) {
+		t.Helper()
+
+		ctx := t.Context()
+		storageInstance := storage.NewStorage(driverInstance)
+		typed := integrity.NewTypedBuilder[IntegrationStruct](storageInstance).
+			WithPrefix("/test").
+			Build()
+
+		_, err := typed.Get(ctx, "my-object")
+		require.Error(t, err, "Get should fail")
+	})
+}
+
+func TestTypedIntegration_Range(t *testing.T) {
+	executeOnStorage(t, func(t *testing.T, driverInstance driver.Driver) {
+		t.Helper()
+
+		fillDataOptions := FillDataOptions{
+			Prefix: "/test",
+			Names:  []string{"object1", "object2"},
+			Records: []IntegrationStruct{
+				{
+					Name:  "obj1",
+					Value: 100,
+				},
+				{
+					Name:  "obj2",
+					Value: 200,
+				},
+			},
+			Hashers: nil,
+			Signers: nil,
+		}
+
+		fillData(t, driverInstance, fillDataOptions)
+
+		ctx := t.Context()
+		storageInstance := storage.NewStorage(driverInstance)
+		typed := integrity.NewTypedBuilder[IntegrationStruct](storageInstance).
+			WithPrefix("/test").
+			Build()
+
+		results, err := typed.Range(ctx, "")
+		require.NoError(t, err)
+		require.Len(t, results, 2)
+
+		foundObj1 := false
+		foundObj2 := false
+
+		for _, result := range results {
+			require.NoError(t, result.Error)
+			require.True(t, result.Value.IsSome())
+
+			val, ok := result.Value.Get()
+			require.True(t, ok)
+
+			if val.Name == "obj1" && val.Value == 100 {
+				foundObj1 = true
+
+				assert.Equal(t, "object1", result.Name)
+			} else if val.Name == "obj2" && val.Value == 200 {
+				foundObj2 = true
+
+				assert.Equal(t, "object2", result.Name)
+			}
+		}
+
+		assert.True(t, foundObj1, "object1 not found in results")
+		assert.True(t, foundObj2, "object2 not found in results")
```

Comment:
It looks like the https://pkg.go.dev/github.com/stretchr/testify/require#ElementsMatch call could be used here. Something like:

```
require.ElementsMatch(results, Result[]{Results{expected1}, Results{expected2}})
```

But I'm not sure. Please, try to use the assertion instead the code.